File: v1/src/main/java/com/api/v1/controllers/payment/AllPaymentsDeletionController.java
package com.api.v1.controllers.payment;

import com.api.v1.services.payment.AllPaymentsDeletionService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;
import reactor.core.publisher.Mono;

@RestController
@RequestMapping("api/v1/payments")
public class AllPaymentsDeletionController {

    @Autowired
    private AllPaymentsDeletionService service;

    @DeleteMapping
    @ResponseStatus(value = HttpStatus.NO_CONTENT)
    public Mono<Void> deleteAll() {
        return service.deleteAll();
    }

    @DeleteMapping("{orderNumber}")
    @ResponseStatus(value = HttpStatus.NO_CONTENT)
    public Mono<Void> deleteByOrderNumber(@PathVariable String orderNumber) {
        return service.deleteByOrderNumber(orderNumber);
    }

}

File: v2/src/main/kotlin/com/api/v2/payment/controllers/AllPaymentsDeletionController.kt
package com.api.v2.payment.controllers

import AllPaymentsDeletionService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus.*
import org.springframework.web.bind.annotation.*

@RequestMapping("api/v2/payments")
@RestController
class AllPaymentsDeletionController {

    @Autowired
    lateinit var service: AllPaymentsDeletionService

    @ResponseStatus(value = NO_CONTENT)
    @DeleteMapping
    suspend fun deleteAll() {
        return service.deleteAll()
    }

    @ResponseStatus(value = NO_CONTENT)
    @DeleteMapping("{orderNumber}")
    suspend fun deleteByOrderNumber(@PathVariable orderNumber: String) {
        return service.deleteByOrderNumber(orderNumber)
    }

}